Added support for huge payloads

pmccarren opened the pull request.

fiddur commented:
Looks like a good start. Add some tests as well!

pmccarren commented:
@fiddur We're good to go! Struggling to find a good way to test it w/o going super overboard. The testEcho method inherently stresses it when sending payloads larger than 4kb.. So I think it's sufficient, however I am open to suggestions!

fiddur commented on lib/Base.php:
```diff
@@ -26,6 +27,11 @@ public function setTimeout($timeout) {
     }
   }
 
+  public function setFragmantSize($fragment_size) {
```
The method name is misspelled.

pmccarren replied:
Ah shoot! I suppose testing this method in the first place would have caught the typo! Adding tests in now

fiddur commented:
Could you try making a test like the echoTest, with fragment size set to something low, like 10 bytes?
The spec says “A sender MAY create fragments of any size for non-control messages.”

And, you could test setFragmentSize, as $options is actually a public property :)

The spec also says control frames can be mixed in, but I don't think that needs to be considered in this pr.  The only control frames handles are 'close', and if you close between fragments, I guess you don't receive the whole payload anyway!

pmccarren commented:
I introduced a class named 'Websocket\Tests\ClientTracker' that acts as a way to track the fragment operations. I also modified testEcho to use a random fragment size, and assert that the proper number of fragments were sent. As always, I am open to any and all suggestions / recommendations!

pmccarren commented:
@fiddur Appears the TravisCI instance erred.. Can you kick off a rebuild? Running locally, there aren't any phpunit errors, so I suppose it was a just a one off fluke.

fiddur commented:
Some maintenance on travis right now; rebuild scheduled...

fiddur commented on tests/ClientTracker.php:
```diff
@@ -0,0 +1,53 @@
+	public $fragment_count;
+
+	public function __construct($uri, $options = array()) {
+		$this->fragment_count = [
```
Make it 5.3 compatible :)  
https://travis-ci.org/Textalk/websocket-php/jobs/59414709

pmccarren commented:
@fiddur 5.3 compatible now! :)

fiddur commented:
Well, looks good.

Thank's for the improvement!
